### ISSUE TITLE
Extract dephosphorylation

### ISSUE THREAD

**Issue by bgyori** [clulab/reach#113]
As far as I see, there is currently no mechanism to extract dephosphorylation events. For instance, parsing the sentence "The phosphatase DUSP6 dephosphorylates MAPK1 at threonine 185 and tyrosine 187."  returns no events.

**Comment by MihaiSurdeanu** [comment]
Ben,
This is trivial to add. We just didn't know this was relevant.
Mihai

**Comment by bgyori** [comment]
Yes, I think it is relevant - it is as important as phosphorylation but less well understood. Still, it is commonly discussed in the literature and in RAS biology in particular.

**Comment by hickst** [comment]
Hi Ben,

While we're adding dephosphorylation, which other PTMs events might be important to add? Any or all of the following? Are there ones I've missed which should be included? Thanks for your help!

Deacetylation
Defarnesylation
Deglycosylation
Dehydrolysis
Dehydroxylation
Demethylation
Deribosylation
Desumoylation
Deubiquitination

**Comment by bgyori** [comment]
I think that list is pretty comprehensive. 99% of the time our group will be interested in Dephosphorylation only but - as long as it's not a lot of extra work - the others listed can definitely be included. I can't think of any further ones right now that are important.

**Comment by hickst** [comment]
Thanks Ben! We'll get to work on adding these.

**Comment by myedibleenso** [comment]
## Proposal
Here are the components I know that we'll need to change:

- [ ] the [`taxonomy`](https://github.com/clulab/reach/blob/f0db436a0efba480b1b6440a2ccfac4a745c478e/src/main/resources/edu/arizona/sista/reach/biogrammar/taxonomy.yml#L26-L39) should be modified to include De* events under `SimpleEvents`.  Ideally, these should be nested under some category under `SimpleEvent`.  Doing so will make things easier in assembly...
  - ex.
   ```
    - SimpleEvent:
      - RemovalEvent:
        -Dephosphorylation
        ...
      - Phosphorylation
     ....
  ```

- [ ] the `!word=/(?i)^de/` pattern in [simple-event_template.yml](https://github.com/clulab/reach/blob/f0db436a0efba480b1b6440a2ccfac4a745c478e/src/main/resources/edu/arizona/sista/reach/biogrammar/events/simple-event_template.yml) should be made a variable.  For `RemovalEvent`s (there's probably a better name for these guys, the value of this variable with be set to `word=/(?i)^de/` in [events_master.yml](https://github.com/clulab/reach/blob/f0db436a0efba480b1b6440a2ccfac4a745c478e/src/main/resources/edu/arizona/sista/reach/biogrammar/events_master.yml).  For the other `SimpleEvent`s, it would be set to `!word=/(?i)^de/` in [events_master.yml](https://github.com/clulab/reach/blob/f0db436a0efba480b1b6440a2ccfac4a745c478e/src/main/resources/edu/arizona/sista/reach/biogrammar/events_master.yml).
- [ ] `getModificationLabel` in [DarpaActions](https://github.com/clulab/reach/blob/f0db436a0efba480b1b6440a2ccfac4a745c478e/src/main/scala/edu/arizona/sista/reach/DarpaActions.scala#L386-L398) will need to be modified to include these "PTMs".  Alternatively, you could create a new kind of [`Modification`](https://github.com/clulab/reach/blob/f0db436a0efba480b1b6440a2ccfac4a745c478e/src/main/scala/edu/arizona/sista/reach/mentions/Modifications.scala) that is virtually identical to [`PTM`](https://github.com/clulab/reach/blob/f0db436a0efba480b1b6440a2ccfac4a745c478e/src/main/scala/edu/arizona/sista/reach/mentions/Modifications.scala#L23-L35), but you'd have to essentially duplicate several rules in [`modifications.yml`](https://github.com/clulab/reach/blob/f0db436a0efba480b1b6440a2ccfac4a745c478e/src/main/resources/edu/arizona/sista/reach/biogrammar/modifications/modifications.yml#L5-L74).
- [ ] Add tests under [`TestModifications`](https://github.com/clulab/reach/blob/f0db436a0efba480b1b6440a2ccfac4a745c478e/src/test/scala/edu/arizona/sista/reach/TestModifications.scala) and [`TestTemplaticSimpleEvents`](https://github.com/clulab/reach/blob/f0db436a0efba480b1b6440a2ccfac4a745c478e/src/test/scala/edu/arizona/sista/reach/TestTemplaticSimpleEvents.scala)
- [ ] Modify the Exporter to handle these cases?  I'm not sure where this is...

**Comment by hickst** [comment]
Thanks for the great proposal comment!
I'm a bit hesitant about the taxonomy idea, though, as I think these events are semantically at the same level as their counterparts: they represent the "inverse" chemical reaction. As I understand it, many of these inverse reactions are continually running in parallel with the forward reactions as the local environment tries to maintain equilibrium. So, if our taxonomy represents real-world semantics, then the inverse reactions would seem to belong at the same level as the forward reactions.

**Comment by myedibleenso** [comment]
You're giving too much credit to our taxonomy.  The taxonomy change I suggested simplifies our lives downstream.

**Comment by myedibleenso** [comment]
You could add `Phosphorylation`, etc. under their own category (`AdditionEvent`?  I'm not very good at naming these) if the symmetry helps you sleep at night.

**Comment by hickst** [comment]
Since any object can be classified along a wide variety of axises, the relevance of any taxonomy depends on ​*how you intend to use it*​. Since subcategorization will be useful for assembly, the SimpleEvents have been divided into AdditionEvents and RemovalEvents.

This issue closed by pull request #159
